Src/CSharpier/DocTypes/Doc.cs

```diff
@@ -8,6 +11,164 @@ public class Doc
         }
 
         public static NullDoc Null { get; } = NullDoc.Instance;
+
+        public static Doc BreakParent => new BreakParent();
+
+        public static HardLine HardLine => new();
+
+        // TODO all of the Line types can probably turn into proper classes, and be the same instance by type
+        public static LiteralLine LiteralLine => new();
+
+        public static LineDoc Line => new() { Type = LineDoc.LineType.Normal };
+
+        public static Trim Trim => Trim.Instance;
+
+        public static LineDoc SoftLine =>
+            new() { Type = LineDoc.LineType.Soft };
+
+        public static HardLineIfNoPreviousLine HardLineIfNoPreviousLine =>
+            new();
+
+        public static LeadingComment LeadingComment(
+            string comment,
+            CommentType commentType
+        ) {
+            return new() { Type = commentType, Comment = comment };
+        }
+
+        public static TrailingComment TrailingComment(
+            string comment,
+            CommentType commentType
+        ) {
+            return new() { Type = commentType, Comment = comment, };
+        }
+
+        public static Concat Concat(List<Doc> contents)
+        {
+            return new(CleanContents(contents));
+        }
+
+        public static Concat Concat(params Doc[] contents)
+        {
+            return new(CleanContents(contents.ToList()));
+        }
+
+        public static Doc Join(Doc separator, IEnumerable<Doc> array)
+        {
+            var docs = new List<Doc>();
+
+            var list = array.ToList();
+
+            if (list.Count == 1)
+            {
+                return list[0];
+            }
+
+            for (var x = 0; x < list.Count; x++)
+            {
+                if (x != 0)
+                {
+                    docs.Add(separator);
+                }
+
+                docs.Add(list[x]);
+            }
+
+            return Concat(docs);
+        }
+
+        public static ForceFlat ForceFlat(List<Doc> contents)
+        {
+            return new()
+            {
+                Contents = contents.Count == 0 ? contents[0] : Concat(contents),
+            };
+        }
+
+        public static ForceFlat ForceFlat(params Doc[] contents)
+        {
+            return new()
+            {
+                Contents = contents.Length == 0
+                    ? contents[0]
+                    : Concat(contents),
+            };
+        }
+
+        public static Group Group(List<Doc> contents)
+        {
+            return new()
```

Comment:
Minor point, but I think these might look better with the expression body syntax